File: app/src/main/java/arunkbabu90/popmovies/adapter/CategoryAdapter.java
package arunkbabu90.popmovies.adapter;

import android.content.Context;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import arunkbabu90.popmovies.R;
import arunkbabu90.popmovies.fragments.FavouritesFragment;
import arunkbabu90.popmovies.fragments.PopularFragment;
import arunkbabu90.popmovies.fragments.TopRatedFragment;

/*
 * Code Adapted from Android Basics: Multi-screen course Udacity
 * https://classroom.udacity.com/courses/ud839
 */
public class CategoryAdapter extends FragmentPagerAdapter
{
    private final Context mContext;

    public CategoryAdapter(Context context, FragmentManager fragmentManager) {
        super(fragmentManager);
        mContext = context;
    }

    @Override
    public Fragment getItem(int position) {
        // Populate the fragments to the correct locations for popular and top rated movies in the view pager
        switch (position)
        {
            case 0:
                return new PopularFragment();
            case 1:
                return new TopRatedFragment();
            case 2:
                return new FavouritesFragment();
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return 3;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        // Set the tab title based on the respective positions of the fragments
        switch (position)
        {
            case 0:
                return mContext.getString(R.string.tab_title_popular);
            case 1:
                return mContext.getString(R.string.tab_title_top_rated);
            case 2:
                return mContext.getString(R.string.tab_title_favourites);
            default:
                return null;
        }
    }
}

File: app/src/main/java/arunkbabu90/popmovies/ui/adapter/CategoryAdapter.kt
package arunkbabu90.popmovies.ui.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import arunkbabu90.popmovies.ui.fragment.NowPlayingFragment
import arunkbabu90.popmovies.ui.fragment.PopularFragment
import arunkbabu90.popmovies.ui.fragment.SearchFragment
import arunkbabu90.popmovies.ui.fragment.TopRatedFragment
import org.jetbrains.annotations.NotNull

class CategoryAdapter(@NotNull fm: FragmentManager, @NotNull lifecycle: Lifecycle)
    : FragmentStateAdapter(fm, lifecycle) {

    override fun getItemCount(): Int = 4

    override fun createFragment(position: Int): Fragment {
        return when (position) {
            0 -> NowPlayingFragment()
            1 -> PopularFragment()
            2 -> TopRatedFragment()
            3 -> SearchFragment()
            else -> NowPlayingFragment()
        }
    }
}